src/hotspot/share/utilities/lineReader.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+#include "runtime/os.hpp"
+#include "utilities/lineReader.hpp"
+
+LineReader::LineReader() {
+  _file = nullptr;
+  _is_oom = false;
+  _buffer = nullptr;
+  _buffer_len = 0;
+}
+
+LineReader::~LineReader() {
+  if (_buffer != nullptr) {
+    os::free(_buffer);
+  }
+}
+
+void LineReader::init(FILE* file) {
+  _file = file;
+  _buffer_len = 16; // start at small size to test expansion logic
```

Comment:
Maybe set the `_buffer_len` to a larger value (256?) for non-debug build?

Reply:
I changed to 

```
_buffer_len = DEBUG_ONLY(16) NOT_DEBUG(4096);
```